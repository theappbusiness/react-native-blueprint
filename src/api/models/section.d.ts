export interface Section {
  title: string;
  body_elements: any[];
}
