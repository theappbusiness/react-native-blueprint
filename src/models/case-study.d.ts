import {Section} from './section';

export interface CaseStudy {
  id: string;
  title: string;
  teaser: string;
  client: string;
  vertical: string;
  hero_image: string;
  sections: Section[];
}
