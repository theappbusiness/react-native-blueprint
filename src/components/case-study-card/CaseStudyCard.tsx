import React from 'react';
import {TouchableOpacity, Text, Image} from 'react-native';
import {CaseStudy} from 'api/models/case-study';
import {styles} from './styles';

interface Props {
  item: CaseStudy;
  onPress?: () => void;
}

const CaseStudyCard = ({item, onPress}: Props) => {
  return (
    <TouchableOpacity
      testID="CaseStudyCard"
      onPress={onPress}
      style={styles.card}>
      <Image
        source={{uri: item.hero_image}}
        style={styles.heroImage}
        resizeMode="cover"
      />
      <Text style={styles.txtTitle}>{item.title}</Text>
      <Text style={styles.txtTeaser}>{item.teaser}</Text>
    </TouchableOpacity>
  );
};

export default CaseStudyCard;
