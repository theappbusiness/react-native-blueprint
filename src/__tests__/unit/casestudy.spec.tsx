import * as React from 'react';
import renderer from 'react-test-renderer';
import {render, cleanup, fireEvent} from '@testing-library/react-native';
import {CaseStudy} from 'api/models/case-study';
import CaseStudyCard from 'components/case-study-card/CaseStudyCard';
import {Text} from 'react-native';

// import {NavigationContainer} from '@react-navigation/native';
// import {render, waitFor} from '@testing-library/react-native';
// import Navigator from 'navigation/RootNavigation';

afterEach(cleanup);

let caseStudy: CaseStudy;

beforeEach(async () => {
  caseStudy = {
    id: '1',
    title: 'Test Case Study',
    sections: [],
  };
});

describe('Basic unit tests', () => {
  it('Renders CaseStudyCard snapshot as expected', () => {
    const tree = renderer.create(<CaseStudyCard item={caseStudy} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // probably not required after snapshot testing passed. Still keeping it for reference
  it("CaseStudyCard's title is bold", () => {
    const rendererInstance = renderer.create(
      <CaseStudyCard item={caseStudy} />,
    );

    const root = rendererInstance.root;
    const title = root.findAllByType(Text)[0];

    const boldStyle = {
      fontWeight: 'bold',
    };
    expect(title.props.style).toMatchObject(boldStyle);
  });

  it('given a CaseStudyCard, user can click on it to invoke onPress', () => {
    const mockFn = jest.fn();

    const {getByTestId} = render(
      <CaseStudyCard item={caseStudy} onPress={mockFn} />,
    );

    fireEvent.press(getByTestId('CaseStudyCard'));
    expect(mockFn).toBeCalled();
  });

  // it('renders the correct screen', async () => {
  //   const {getByTestId} = render(
  //     <NavigationContainer>
  //       <Navigator />
  //     </NavigationContainer>,
  //   );
  //   await waitFor(() => getByTestId('CaseStudiesScreen'));
  // });
});
