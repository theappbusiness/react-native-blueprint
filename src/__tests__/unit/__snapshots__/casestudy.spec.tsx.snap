// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic unit tests Renders CaseStudyCard snapshot as expected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#FFFFFF",
      "elevation": 3,
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 16,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
  testID="CaseStudyCard"
>
  <Image
    resizeMode="cover"
    source={
      {
        "uri": undefined,
      }
    }
    style={
      {
        "height": 140,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginHorizontal": 16,
        "marginTop": 16,
      }
    }
  >
    Test Case Study
  </Text>
  <Text
    style={
      {
        "color": "#737373",
        "marginHorizontal": 16,
        "marginTop": 4,
      }
    }
  />
</View>
`;
